tests/kernel/run_ltp.pm

```diff
@@ -339,6 +339,34 @@ sub upload_tcpdump {
     select_console($old_console) if defined($old_console);
 }
 
+sub upload_oprofile {
+    my $self = shift;
+    my $pid = $self->{oprofile_pid};
+    my $old_console;
+
+    $self->{oprofile_pid} = undef;
+
+    if ($self->{timed_out}) {
+        $old_console = current_console();
+        select_console('root-console');
+
+        unless (defined(script_run("timeout 20 sh -c \"kill -s INT $pid && while [ -d /proc/$pid ]; do usleep 100000; done\""))) {
+            select_console($old_console, await_console => 0);
+            return;
+        }
+    }
+    else {
+        assert_script_run("kill -s INT $pid && wait $pid");
+    }
+
+    assert_script_run('pushd /tmp');
+    assert_script_run("tar cjf /tmp/ltp_oprofile_data.tar.bz2 ltp_oprofile");
+    assert_script_run('popd');
+    upload_logs("/tmp/ltp_oprofile_data.tar.bz2");
+    upload_logs("/tmp/ltp_oprofile.log");
```

Comment:
Should it use a .txt extension to make use of openQA internal viewer?

Reply:
Done, one more verification run restarted.